frontend/src/utils/permissions.js

```diff
@@ -0,0 +1,31 @@
+//
+// Copyright (c) 2019 by SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+import filter from 'lodash/filter'
+import includes from 'lodash/includes'
+import isEmpty from 'lodash/isEmpty'
+
+function canI ({ resourceRules = [] } = {}, verb, apiGroup, resouce, resourceName) {
+  let filtered = filter(resourceRules, ({ apiGroups }) => includes(apiGroups, apiGroup) || includes(apiGroups, '*'))
+  filtered = filter(filtered, ({ resources }) => includes(resources, resouce) || includes(resources, '*'))
+  filtered = filter(filtered, ({ verbs }) => includes(verbs, verb) || includes(verbs, '*'))
+  filtered = filter(filtered, ({ resourceNames }) => isEmpty(resourceNames) || includes(resourceNames, resourceName) || includes(resourceNames, '*'))
```

Comment:
isEmpty only for resourceNames?

Reply:
done, added reuse function
Yes, isEmpty  check only for resource names.